src/replication.c

```diff
@@ -2983,6 +3517,12 @@ int cancelReplicationHandshake(int reconnect) {
     {
         undoConnectWithMaster();
         server.repl_state = REPL_STATE_CONNECT;
+    } else if (server.repl_state == REPL_SEC_CONN_RECEIVE_PSYNC_REPLY || 
+               server.repl_state == REPL_SEC_CONN_SEND_PSYNC){
+        /* As we cancel the sync, the rdb connection state is no longer relevant */
+        server.repl_state = REPL_STATE_TRANSFER;
+        replicationAbortSyncTransfer();
```

Comment:
I don't think we should artificially change the state to transfer to avoid the server assert in the function. Also, this functions documentation is "Don't call this function directly", so it seems like we should honor that.

Reply:
I agree regarding the artificial change, I will fix that.
In regards to calling replicationAbortSyncTransfer directly, the documentation also states "...use cancelReplicationHandshake() instead.". By calling replicationAbortSyncTransfer from within cancelReplicationHandshake, I feel like I honor the request 😄